### ISSUE TITLE
Ticking Entity Crash when re-entering a world 1.21.1 Fabric

### ISSUE THREAD

**Issue by Yoni664** [Goby56/wakes#108]
**Describe the bug**
When re-entering a world the game crashes due to ticking entity error, I used Neruina to track the crashes and it points to a problem with Wakes and Puzzleslib.
The crash only occurs when you want to re-enter the world, what I mean is that when you start minecraft for the first time you can enter the world and play normally, only when you exit to the main menu and re-enter the world is when the crash occurs.
From what I noticed the crash is generated by entities that are in water when loading the world, even items that are floating in the water generate the crash.

**Expected behavior**
Re-entering the world without problems

**To Reproduce**
Open minecraft
Create or join a world
Exit the world to the main menu
Re-enter the world

**Screenshots**
![image](https://github.com/user-attachments/assets/c8ea7282-6029-4127-bba0-54403ba34d55)
Image from the chat message of Neruina

**Hardware and OS**
If the bug is related to rendering, please provide at least which graphics card and OS you are using.

**Versions**
- Minecraft version 1.21.1
- Mod loader and version (e.g fabric or quilt) Fabric 0.16.5
- Wakes version 0.3.0
- Fabric API (or quilt equivalent) version 0.104.0

**Additional information**
Link to Neruina Auto Report [puzzleslib, wakes] (25 sept 2024 - 17:47:43):
https://github.com/Bawnorton/NeruinaAutoReports/issues/203
Crash Log:
[crash-2024-09-25_17.41.09-server.txt](https://github.com/user-attachments/files/17139751/crash-2024-09-25_17.41.09-server.txt)

**Comment by Yoni664** [comment]
Another crash report from Neruina. This time there were so many entities that a CTD simply occurred:
[crash-2024-09-25_18.22.10-server.txt](https://github.com/user-attachments/files/17139893/crash-2024-09-25_18.22.10-server.txt)

**Comment by Yoni664** [comment]
Using Wakes 0.2.5 on 1.21.1 seems to work without any problems.

**Comment by AlienXtream** [comment]
getting this on 1.20.1 too so its likely a core issue. seems to be related to something about the wake sim ticking. here's the crash from my version

[crash-2024-09-28_21.22.11-server.txt](https://github.com/user-attachments/files/17174211/crash-2024-09-28_21.22.11-server.txt)

**Comment by bachleon** [comment]
same ticking entity for me with the same re-loading behaviour.  I can enter the game from the launcher once with no issues whatsoever, but quitting to menu then re-loading a world crashes every time.  I manually removed mods 1 at a time with no change until I removed wakes, which has fixed the issue entirely.

[crash-2024-10-06_08.45.03-server.txt](https://github.com/user-attachments/files/17269657/crash-2024-10-06_08.45.03-server.txt)
